developer/frameworks/articles/data-engine/01-data-definition-api/01-data-definition-api-operations.markdown

```diff
@@ -94,10 +105,85 @@ To delete a `DEDataDefinition`, pass a `DEDataDefinitionDeleteRequest` to
 Required parameter: `long dataDefinitionId`
 Optional parameters: none
 
-<!-- API NOT READY FOR DOCS YET -->
-<!--
-## Defining Permissions on Data Definitions
+## Granting and Revoking Permissions on Data Definitions
+
+The permissions operations for `DEDataDefinition`s work similarly to the other
+operations. Pass a `DEDataDefinition*PermissionsRequest` to
+`DEDataDefinitionService.execute`. There are permissions for granting and
+revoking permissions on Data Definitions.
+
+### Granting and Revoking Permission to Add Data Definitions and Manage Permissions
+
+`DEDataDefinitionSavePermissionsRequest` is for granting one or more Roles
+permission to add a new Data Definition to the system, and/or the ability to
+manage User permissions on Data Definitions.
+
+Required parameters: 
+- `long companyId`
+- `long scopeGroupId`
+- `String[] roleNames`
+
+Optional parameters: none
+
+Optional builder methods:
+- `allowAddDefinition()` to grant Add permissions
+- `allowDefinePermissions()` to grant permissions to grant permissions
+<!-- Find a better way to say the above -->
+
+`DEDataDefinitionDeletePermissionsRequest` is for revoking the same permissions
+granted by the `DeDataDefinitionSavePermissionsRequest`. 
+
+Required parameters: 
+- `long companyId`
+- `long scopeGroupId`
+- `String[] roleNames`
+
+Optional parameters: none
+
+Once the `DEDataDefinitionDeletePermissionsRequest` is executed, all specified
+Roles lose both permissions simultaneously: Role Users can no long add Data
+Definitions or manage permissions for Data Definitions. 
+
+### Granting and Revoking Permission to View, Update, or Delete a specific Data Definition
+
+`DEDataDefinitionSaveModelPermissionsRequest` is for granting permission to
+update, view, and delete a specific Data Definition already in the system. 
+
+Required parameters:
+- `long companyId`
+- `long groupId`
+- `long scopeUserId`
+- `long scopeGroupId`
+- `long deDataDefinitionId`
+- `String[] roleNames`
+
+Optional parameters: none
+
+Optional builder methods:
+- `allowDelete()` to grant deletion permission
+- `allowUpdate()` to grant update permission
+- `allowView()` to grant view permission
+
+Specify more than one builder method to grant mutliple permissions
+simultaneously.
+
+Here's an example request builder that grants Site Members all three permissions:
+
+`DEDataDefinitionDeleteModelPermissionsRequest` is for revoking permission to
+view, update, or delete two permissions: the permission to define new Data
+Definition permissions for a User, and the permission to add a new entity
+registry in the system. If a user choose to make this type of delete request,
+all the roles listed in the request will lose both permissions at the same
+time. 
```

Comment:
This paragraph is not right. The difference of the DeleteModelPermission to the DeletePermission is that the user can specify in the "actionIDs" parameter which permission will revoke, while in the other one the user loses all permissions at the same time without specifying anything.

I wrote in this way, but you can change if you want to:
DEDataDefinitionDeleteModelPermissionsRequest is for revoking permission to view, update, and/or delete a specific data definition already in the system. The user can insert one or more permissions that wants to revoke in the “actionIds” parameter in the request.

Reply:
Ahh, my apologies. I copied this into the wrong place and didn't finish editing it. I'll fix it.